app/src/main/kotlin/io/intrepid/skotlinton/base/BasePresenter.kt

```diff
@@ -8,57 +11,31 @@ import io.reactivex.Observable
 import io.reactivex.Scheduler
 import io.reactivex.Single
 import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
 
-abstract class BasePresenter<V : BaseContract.View>(protected var view: V?, configuration: PresenterConfiguration) : BaseContract.Presenter {
+abstract class BasePresenter<S : BaseScreen>(protected var screen: S, configuration: PresenterConfiguration) : LifecycleObserver {
 
-    protected val ioScheduler: Scheduler = configuration.ioScheduler
-    protected val uiScheduler: Scheduler = configuration.uiScheduler
+    private val ioScheduler: Scheduler = configuration.ioScheduler
+    private val uiScheduler: Scheduler = configuration.uiScheduler
     protected val userSettings: UserSettings = configuration.userSettings
     protected val restApi: RestApi = configuration.restApi
     protected val crashReporter: CrashReporter = configuration.crashReporter
 
     protected val disposables: CompositeDisposable = CompositeDisposable()
 
-    private var isViewBound = false
-
-    override fun onViewCreated() {
-
-    }
-
-    override fun bindView(view: BaseContract.View) {
-        @Suppress("UNCHECKED_CAST")
-        this.view = view as V
-        if (!isViewBound) {
-            onViewBound()
-            isViewBound = true
-        }
-    }
-
-    open protected fun onViewBound() {
-
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    fun onCreateScreenLifecycle() {
+        Timber.v("Presenter onCreateScreenLifecycle: $this")
+        screenCreated()
     }
 
-    override fun unbindView() {
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onStopScreenLifecycle() {
         disposables.clear()
-        this.view = null
-
-        if (isViewBound) {
-            onViewUnbound()
-            isViewBound = false
-        }
-    }
-
-    open protected fun onViewUnbound() {
-
+        Timber.v("Presenter onStopScreenLifecycle: $this")
```

Comment:
Question: Does changing the presenter's view/screen from `S?` to `S` mean that, since we don't set it to null here anymore, we are now assuming that *all* references to the view/screen in the presenter are contained inside a subscription that has been added to `disposables`?

Reply:
it seemed like have a nullable view was a safety feature in case that didn't happen

Reply:
I feel like we want a situation where the `screen` (view) can be null or some kind of stub to guard against any presenter calls after `onStop()`. After some experimentation, I noticed as long as we're correctly instantiating the `BasePresenter`, we don't have to worry about the `screen` (view) being in a null state. I'm very open to suggestions on how to make this more air tight and less magic

Reply:
Also to note:
In the current skeleton project, we're binding the screen in `onStart()`, but this has no effect since the view is initialized in the constructor. This was from the cmnd+alt+opt+K conversion.

Reply:
One consideration with this change is that since the view is now entirely tied to the presenter, it would not be possible to add mechanisms to preserve the presenter during Activity recreate since it cannot be reattached to another view and would cause a leak